pkg/process/config/config_nix_test.go

```diff
@@ -25,6 +25,12 @@ import (
 
 var secretScriptBuilder sync.Once
 
+// secretsTestRestoreConfig is an alternate version of restoreGlobalConfig designed for secrets tests
```

Comment:
From this comment, it's a bit unclear why `cgroup` provider affects these tests.

Reply:
I added some more detail. Let me know if that helps clear things up

Reply:
Ok, it makes a bit more sense, concern is it's quite hard to see why we would need two separate helpers, the "allowing tests to pass" sounds suspicious. Can we just have `resetConfigDatadog` which doesn't use a global, and use it in both places?

Reply:
Per our discussion, we decided that the best solution is to pass `canAccessContainers`, to `NewAgentConfig` and get rid of `secretsTesetRestoreConfig` entirely. That way it is clear that these tests are not designed to access containers, and we don't have to worry about messing with any confusing cgroup provider settings.